test/integration/etcd_storage_path_test.go

```diff
@@ -0,0 +1,1079 @@
+		if err != nil {
+			kindSeen.Insert(kind)
+			if kindWhiteList.Has(kind) {
+				// t.Logf("skipping test for %s from %s because its GVK %s is whitelisted and has no mapping", kind, pkgPath, gvk)
+			} else {
+				t.Errorf("no mapping found for %s from %s but its GVK %s is not whitelisted", kind, pkgPath, gvk)
+			}
+			continue
+		}
+
+		gvResource := gvk.GroupVersion().WithResource(mapping.Resource)
+		etcdSeen[gvResource] = empty{}
+
+		testData, hasTest := etcdStorageData[gvResource]
+		_, isEphemeral := ephemeralWhiteList[gvResource]
+
+		if !hasTest && !isEphemeral {
+			t.Errorf("no test data for %s from %s.  Please add a test for your new type to etcdStorageData.", kind, pkgPath)
+			continue
+		}
+
+		if hasTest && isEphemeral {
+			t.Errorf("duplicate test data for %s from %s.  Object has both test data and is ephemeral.", kind, pkgPath)
+			continue
+		}
+
+		if isEphemeral { // TODO it would be nice if we could remove this and infer if an object is not stored in etcd
+			// t.Logf("Skipping test for %s from %s", kind, pkgPath)
+			ephemeralSeen[gvResource] = empty{}
+			delete(etcdSeen, gvResource)
+			continue
+		}
+
+		if len(testData.expectedEtcdPath) == 0 {
+			t.Errorf("empty test data for %s from %s", kind, pkgPath)
+			continue
+		}
+
+		shouldCreate := len(testData.stub) != 0 // try to create only if we have a stub
+
+		var input map[string]interface{}
+		if shouldCreate {
+			if input, err = jsonToMap(testData.stub); err != nil || isZeroMap(input) {
+				t.Errorf("invalid test data for %s from %s: %s", kind, pkgPath, err)
+				continue
+			}
+		}
+
+		func() { // forces defer to run per iteration of the for loop
+			all := &[]cleanupData{}
+			defer func() {
+				if !t.Failed() { // do not cleanup if test has already failed since we may need things in the etcd dump
+					if err := client.cleanup(all); err != nil {
+						t.Fatalf("failed to clean up etcd: %#v", err)
+					}
+				}
+			}()
+
+			if err := client.createPrerequisites(mapper, testNamespace, testData.prerequisites, all); err != nil {
+				t.Errorf("failed to create prerequisites for %s from %s: %#v", kind, pkgPath, err)
+				return
+			}
+
+			if shouldCreate { // do not try to create items with no stub
+				if err := client.create(testData.stub, testNamespace, mapping, all); err != nil {
+					t.Errorf("failed to create stub for %s from %s: %#v", kind, pkgPath, err)
+					return
+				}
+			}
+
+			output, err := getFromEtcd(keys, testData.expectedEtcdPath)
+			if err != nil {
+				t.Errorf("failed to get from etcd for %s from %s: %#v", kind, pkgPath, err)
+				return
+			}
+
+			expectedGVK := gvk
+			if testData.expectedGVK != nil {
+				expectedGVK = *testData.expectedGVK
+			}
+
+			actualGVK := getGVK(output)
+			if actualGVK != expectedGVK {
+				t.Errorf("GVK for %s from %s does not match, expected %s got %s", kind, pkgPath, expectedGVK, actualGVK)
+			}
+
+			if !deepDerivativeMap(input, output) {
```

Comment:
I wouldn't do a full diff here, just make sure kind, apiVersion, name are all what we expect